pkg/interfaces/context.go

```diff
@@ -71,17 +69,31 @@ type AppFunctionContext interface {
 	LoggingClient() logger.LoggingClient
 	// EventClient returns the Event client. Note if Core Data is not specified in the Clients configuration,
 	// this will return nil.
-	EventClient() coredata.EventClient
+	EventClient() interfaces.EventClient
 	// CommandClient returns the Command client. Note if Support Command is not specified in the Clients configuration,
 	// this will return nil.
 	CommandClient() interfaces.CommandClient
-	// NotificationsClient returns the Notifications client. Note if Support Notifications is not specified in the
+	// NotificationClient returns the Notification client. Note if Support Notifications is not specified in the
+	// Clients configuration, this will return nil.
+	NotificationClient() interfaces.NotificationClient
+	// SubscriptionClient returns the Subscription client. Note if Support Notifications is not specified in the
+	// Clients configuration, this will return nil.
```

Comment:
in comment, you meant the Subscription is not specified?

Reply:
Actually this is correct. Subscriptions is part of the Support Notifications service, so SubscriptionClient  is dependent on the Support Notifications Client config